enterprise/server/cmd/server/BUILD

```diff
@@ -130,3 +132,27 @@ container_image(
     tags = ["manual"],
     visibility = ["//visibility:public"],
 )
+
+# TODO(sluongng): Verify the layering of the image.
```

Comment:
Worth noting: I am aware that the current pkg_tar setup is wrong.

- It does not pull in the much-needed runfiles of the go binaries
- There are also some layout changes on the final oci image vs the old image that we should address.

Since this PR is already fairly sized and does not break anything, I will be merging this as-is. A subsequent PR will be sent to correct the rules_pkg usage and oci image layout.